runtime/v2/module.go

```diff
@@ -213,27 +215,65 @@ func ProvideEnvironment(
 	memKvService store.MemoryStoreService,
 	headerService header.Service,
 	eventService event.Service,
+	branchService branch.Service,
+	routerBuilder RouterServiceBuilder,
 ) appmodulev2.Environment {
 	return appmodulev2.Environment{
 		Logger:             logger,
-		BranchService:      stf.BranchService{},
+		BranchService:      branchService,
 		EventService:       eventService,
 		GasService:         stf.NewGasMeterService(),
 		HeaderService:      headerService,
-		QueryRouterService: stf.NewQueryRouterService(),
-		MsgRouterService:   stf.NewMsgRouterService([]byte(key.Name())),
+		QueryRouterService: routerBuilder.BuildQueryRouter(),
+		MsgRouterService:   routerBuilder.BuildMsgRouter([]byte(key.Name())),
 		TransactionService: services.NewContextAwareTransactionService(),
 		KVStoreService:     kvService,
 		MemStoreService:    memKvService,
 	}
 }
 
+// ServiceBuilder builds the msg router and query router service during app initialization.
+type RouterServiceBuilder interface {
```

Comment:
Let's add go doc that it is primarily used for testing and that it only override the message router in the module environment.

Reply:
updated